### PR TITLE
Add infinite scroll component, "scroll to top button" and base pokemon page.

### PR DESCRIPTION
**Pull request by NikolasGumenyuk**

### CODE REVIEW

File: src/components/PokemonCard/PokemonCard.tsx [hunk 1]
```diff
@@ -16,16 +20,20 @@ const PokemonCard = ({
...
-
   const [isHover, setIsHover] = useState<boolean>(false);
   const gif = getGifUrl(id);
+  const navigate = useNavigate();
 
-  const capitalizeText = (str: string) => str[0].toUpperCase() + str.slice(1);
-  const pokemonName = capitalizeText(name);
+  const navigateToPokemon = () => {
+    dispatch(setCurrentPokemon(name));
+    navigate(PathName.pokemon);
```
Comment (LobodaAndreyEK) [inline src/components/PokemonCard/PokemonCard.tsx]: Please rewrite it with useParams

File: src/components/PokemonCard/PokemonCard.tsx [hunk 2]
```diff
@@ -16,16 +20,20 @@ const PokemonCard = ({
...
+  };
 
   return (
     <div
+      onClick={() => navigateToPokemon()}
```
Comment (LobodaAndreyEK) [inline src/components/PokemonCard/PokemonCard.tsx]: simplify

File: src/components/PokemonsList/PokemonsList.tsx [hunk 3]
```diff
@@ -10,48 +12,53 @@ const PokemonsList = () => {
...
-    }, 2000);
-
-    return () => clearInterval(interval);
+    window.addEventListener('scroll', () => {
+      if (window.scrollY > 400) {
+        setShowTopBtn(true);
+      } else {
+        setShowTopBtn(false);
+      }
+    });
```
Comment (LobodaAndreyEK) [inline src/components/PokemonsList/PokemonsList.tsx]: When you add some listener, you have to remove it on component unmount

**Review by LobodaAndreyEK — CHANGES_REQUESTED** [review]

**Review by NataliaHanzhuha — APPROVED** [review]
fixed bugs

**Pull request closed without merging**